src/Microsoft.DotNet.ProjectJsonMigration/ProjectExtensions.cs

```diff
@@ -13,59 +13,63 @@ internal static class ProjectExtensions
     {
         public static ProjectType GetProjectType(this Project project)
         {
-            var projectType = ProjectType.Console;
-            if (project.IsWebProject())
+            ProjectType projectType;
+            if (project.IsTestProject)
             {
-                projectType = ProjectType.Web;
+                projectType = ProjectType.Test;
             }
-            else if (project.IsTestProject)
+            else if (project.HasEntryPoint())
             {
-                projectType = ProjectType.Test;
+                if (project.HasDependency(ContainingName(".AspNetCore.")))
+                {
+                    projectType = ProjectType.Web;
+                }
+                else
+                {
+                    projectType = ProjectType.Console;
+                }
+            }
+            else
+            {
+                projectType = ProjectType.Library;
```

Comment:
When creating the projectType variable above, you can just assign ProjectType.Library to it and then you can remove this else clause.